src/commands/force/user/password/generate.ts

```diff
@@ -34,6 +43,12 @@ export class UserPasswordGenerateCommand extends SfdxCommand {
 
   public async run(): Promise<PasswordData[] | PasswordData> {
     this.usernames = (this.flags.onbehalfof as string[]) ?? [this.org.getUsername()];
+    if (this.flags.length > 20 || this.flags.length < 8)
+      throw new SfdxError(messages.getMessage('lengthOutOfBound'), 'lengthOutOfBound');
+    const passwordCondition: PasswordConditions = {
+      length: this.flags.length ? parseInt(this.flags.length, 10) : 13,
```

Comment:
instead of `parseInt` you should use `asNumber` from `@salesforce/ts-types` like `asNumber(this.flags.complexity, 5)`